main.go

```diff
@@ -1,16 +1,37 @@
 package main
 
 import (
-	"net/http"
+	"log"
+	"os"
+	"fmt"
 
+	"net/http"
 	"github.com/gin-gonic/gin"
 	"github.com/gin-gonic/contrib/static"
+
+	"github.com/joho/godotenv"
+	_ "github.com/lib/pq"
+	"database/sql"
 )
 
 func  main()  {
 	//set the router as the default shipped with gin
 	router := gin.Default()
 
+	// connect to postgres
+	err := godotenv.Load()
+	if err != nil {
+		log.Fatal("Error loading .env file")
+	}
+	uri := os.Getenv("DB_URI")
+	db, err := sql.Open("postgres", uri)
+	if err != nil {
+		panic(err)
+	}
+	defer db.Ping()
+	fmt.Println("Connected")
```

Comment:
This will always print connected? Does `defer` throw error if any?

Reply:
Error is thrown  with panic(). If there's error, we never start the server. Rest are in the else block